src/data-workspace/category-combo-table/category-combo-table-header.js

```diff
@@ -0,0 +1,130 @@
+import i18n from '@dhis2/d2-i18n'
+import { TableRowHead, TableCellHead } from '@dhis2/ui'
+import cx from 'classnames'
+import PropTypes from 'prop-types'
+import React from 'react'
+import { useMetadata, selectors } from '../../metadata/index.js'
+import { useActiveCell } from '../data-entry-cell/index.js'
+import styles from './category-combo-table.module.css'
+import { PaddingCell } from './padding-cell.js'
+import { TotalHeader } from './total-cells.js'
+
+export const CategoryComboTableHeader = ({
+    dataElements,
+    renderRowTotals,
+    paddingCells,
+    categoryOptionCombos,
+    categories,
+}) => {
+    const { data: metadata } = useMetadata()
+    const { deId: activeDeId, cocId: activeCocId } = useActiveCell()
```

Comment:
I'm thinking that it would probably benefit the app's architecture and performance if we were to push this form state subscription down to the level of the actual header cell.

Reply:
I don't really think it's necessary to push it further. You could in theory have a `CategoryTableHeaderColumns` and move it there, and basically move the rendered component from `columns.map...`. But that would only save re-rendering of the static components "Category name" and "PaddingCell". Again these are static, with no dynamic props and no DOM-updated are needed.

Reply:
I was mainly concerned about the nested iterations needed to compute the columns, and that fact that they would need to run whenever a the active cell changes. I suggested solving this by pushing the subscription to the active cell down, but I noticed you are now memoizing these columns in the `useCategoryColumns`. I think that's an equally valid solution.